test/raft/unit/test_snapshot.c

```diff
@@ -373,12 +381,45 @@ TEST(snapshot_leader, timeouts, set_up, tear_down, 0, NULL) {
 	return MUNIT_OK;
 }
 
+struct test_fixture {
+	union {
+		struct leader leader;
+		struct follower follower;
+	};
+	/* true when union contains leader, false when it contains follower */
+	bool is_leader;
+	pool_t pool;
+
+	work_op orig_cb;
+	bool work_done;
+};
+
+/* Not problematic because each test runs in a different process. */
+static struct test_fixture global_fixture;
```

Comment:
It should still work with `--no-fork` because the tests are going to be run sequentially. Also, if you are doing that in GDB to not follow the forks, you can instead add a local `.gdbinit` in the project folder with:
```
set follow-fork-mode child
```